addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/dto/HueDevice.java

```diff
@@ -289,14 +294,16 @@ private void setState(State itemState) {
                 HueStateColorBulb c = as(HueStateColorBulb.class);
                 int newV = c.hue + newState.hue_inc;
                 if (newV < 0 || newV > HueStateColorBulb.MAX_HUE) {
-                    throw new ClassCastException();
+                    throw new IllegalArgumentException();
                 }
                 c.colormode = ColorMode.hs;
                 c.hue = newV;
                 command = c.toHSBType();
                 successApplied.put("hue", newState.hue);
             } catch (ClassCastException e) {
                 errorApplied.add("hue_inc");
+            } catch (IllegalArgumentException e) {
```

Comment:
Ideally this should be a multi catch feel free to create an other PR to address that